# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/nsNativeDragTarget.cpp
# Commit: df85fe3d7cb6
# Full Hash: df85fe3d7cb6a219c37e0a80d00f51e07783aec1
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 22 - Make nsIDragService::StartDragSession return the nsIDragSession r=rkraesig,win-reviewers,gstoll,geckoview-reviewers,m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D212298
# ==============================================================================

diff -r 384b025f56d3 -r df85fe3d7cb6 widget/windows/nsNativeDragTarget.cpp
--- a/widget/windows/nsNativeDragTarget.cpp	Thu Jul 04 07:48:11 2024 +0000
+++ b/widget/windows/nsNativeDragTarget.cpp	Thu Jul 04 07:48:12 2024 +0000
@@ -249,7 +249,9 @@
 
   // tell the drag service about this drag (it may have come from an
   // outside app).
-  mDragService->StartDragSession(mWidget);
+  RefPtr<nsDragSession> session =
+      static_cast<nsDragSession*>(mDragService->StartDragSession(mWidget));
+  MOZ_ASSERT(session);
 
   void* tempOutData = nullptr;
   uint32_t tempDataLen = 0;
@@ -264,9 +266,7 @@
     mEffectsPreferred = DROPEFFECT_NONE;
   }
 
-  // Set the native data object into drag service
-  RefPtr<nsDragSession> session =
-      static_cast<nsDragSession*>(mDragService->GetCurrentSession(mWidget));
+  // Set the native data object into drag session
   session->SetIDataObject(pIDataSource);
 
   // Now process the native drag state and then dispatch the event
